contracts/contracts/strategies/LidoWithdrawalStrategy.sol

```diff
@@ -0,0 +1,314 @@
+        address indexed owner,
+        uint256 amountOfStETH,
+        uint256 amountOfShares
+    );
+    event WithdrawalsFinalized(
+        uint256 indexed from,
+        uint256 indexed to,
+        uint256 amountOfETHLocked,
+        uint256 sharesToBurn,
+        uint256 timestamp
+    );
+    event WithdrawalClaimed(
+        uint256 indexed requestId,
+        address indexed owner,
+        address indexed receiver,
+        uint256 amountOfETH
+    );
+
+    struct WithdrawalRequestStatus {
+        /// @notice stETH token amount that was locked on withdrawal queue for this request
+        uint256 amountOfStETH;
+        /// @notice amount of stETH shares locked on withdrawal queue for this request
+        uint256 amountOfShares;
+        /// @notice address that can claim or transfer this request
+        address owner;
+        /// @notice timestamp of when the request was created, in seconds
+        uint256 timestamp;
+        /// @notice true, if request is finalized
+        bool isFinalized;
+        /// @notice true, if request is claimed. Request is claimable if (isFinalized && !isClaimed)
+        bool isClaimed;
+    }
+
+    function requestWithdrawals(uint256[] calldata _amounts, address _owner)
+        external
+        returns (uint256[] memory requestIds);
+
+    function getLastCheckpointIndex() external view returns (uint256);
+
+    function findCheckpointHints(
+        uint256[] calldata _requestIds,
+        uint256 _firstIndex,
+        uint256 _lastIndex
+    ) external view returns (uint256[] memory hintIds);
+
+    function claimWithdrawals(
+        uint256[] calldata _requestIds,
+        uint256[] calldata _hints
+    ) external;
+
+    function getWithdrawalStatus(uint256[] calldata _requestIds)
+        external
+        view
+        returns (WithdrawalRequestStatus[] memory statuses);
+
+    function getWithdrawalRequests(address _owner)
+        external
+        view
+        returns (uint256[] memory requestsIds);
+
+    function finalize(
+        uint256 _lastRequestIdToBeFinalized,
+        uint256 _maxShareRate
+    ) external payable;
+}
+
+/**
+ * @title Lido Withdrawal Strategy
+ * @notice This strategy withdraws ETH from stETH via the Lido Withdrawal Queue contract
+ * @author Origin Protocol Inc
+ */
+contract LidoWithdrawalStrategy is InitializableAbstractStrategy {
+    /// @notice Address of the WETH token
+    IWETH9 private constant weth =
+        IWETH9(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    /// @notice Address of the stETH token
+    IERC20 private constant stETH =
+        IERC20(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
+    /// @notice Address of the Lido Withdrawal Queue contract
+    IStETHWithdrawal private constant withdrawalQueue =
+        IStETHWithdrawal(0x889edC2eDab5f40e902b864aD4d7AdE8E412F9B1);
+    /// @notice Maximum amount of stETH that can be withdrawn in a single request
+    uint256 public constant MaxWithdrawalAmount = 1000 ether;
```

Comment:
nit: For constants, using upper snake case would be nice, not necessary to change at this point